instructionAPI/src/ArchSpecificFormatters.C

```diff
@@ -16,7 +16,25 @@ using namespace Dyninst::InstructionAPI;
 
 ///////// Base Formatter
 
-std::string ArchSpecificFormatter::formatBinaryFunc(std::string left, std::string func, std::string right) {
+std::string ArchSpecificFormatter::getInstructionString(const std::vector<std::string> &operands) const
+{
+    std::string s;
```

Comment:
I would strongly encourage the use of [`boost::algorithm::join_if`](https://www.boost.org/doc/libs/1_82_0/doc/html/boost/algorithm/join_if_1_3_3_9_14_3_1_2.html) here.